universe/Universe.cpp

```diff
@@ -1068,10 +1070,19 @@ namespace {
             target_objects.insert(target_objects.end(), target_set->begin(), target_set->end());
         }
 
+        // log condition scope matches, except for Source
+        if (!(dynamic_cast<const Condition::Source*>(cond))) {
+            std::stringstream ss;
+            ss << "\nGenerated new target set, for Condition: " << cond->Description() << "    targets: (";
```

Comment:
Since this is being logged, rather than shown to the player, this should probably call `cond->Dump(...)` rather than `Description`. Also, why skip the output for `Source` conditions? They're probably almost the shortest, and always having it there makes it easier to serach / visually parse.